app/client/jsx/PuckSelect.jsx

```diff
@@ -8,107 +8,99 @@ class PuckSelect extends Component {
+          if (this.state.rowOpen!==id) { // checks each item in column for whether selected
 
-          if (this.state.rowOpen===id) {
+            let handleClick = () => this.handleClickDesign(id)
+            let image_transparency = 'image'
+            if (this.state.rowOpen!=null && this.state.rowOpen!==id) {
+              image_transparency = 'non-selected-image'
+            }
+            return ( // returns single instance of each nonselected design
+              <PuckBox
+              handleClick={handleClick}
+              key={id}
+              image={puck}
+              style='box'
+              imageStyle={image_transparency}
+              />
+            )
 
-            let variant_array = avatars[this.state.rowOpen].map((color,id) => {
+          } else { // selected design
 
+            let colorway_variants_array = avatars[this.state.rowOpen].map((color,id) => {
               let selected = 'image'
-              // if (id===0) { selected = 'box-selected' }
-              let handleClickColor = () => this.handleClickColor(id)
-
-              // avatar color variant selection
-              return (
+              let handleClick = () => this.handleClickColor(id)
+              return ( // returns all colorways for selected design
                 <PuckBox
-                handleClick={handleClickColor}
+                handleClick={handleClick}
                 key={`puckbox-${id}`}
                 image={color}
                 imageStyle={selected}
                 />
               )
             })
-
-            return (
+            return ( // colorways row in place of selected column item
               <div className='inner'>
-              {variant_array}
+              {colorway_variants_array}
               </div>
             )
 
-          } else {
-
-            let handleClick = () => this.handleClick(id)
-
-            let image_transparency = 'image'
-            if (this.state.rowOpen!=null && this.state.rowOpen!==id) {
-              image_transparency = 'non-selected-image'
-            }
-            // avatar style selection
-            return (
-              <PuckBox
-              handleClick={handleClick}
-              key={id}
-              image={puck}
-              style='box'
-              imageStyle={image_transparency}
-              />
-            )
-
           }
-
         })
       }
 
-      return(
+      return ( // column of singleton nonselected items with at most one of them swapped out for colorway variants row
         <div className='outer'>
         <input className={fade} type="button" onClick={this.onButtonClick} value="Pick your puck"/>
-        {puck_array}
+        {all_avatar_designs_array}
         </div>
       )
 
-    }
+    } else {
 
+      return ( // when avatar state isn't null, collapse all and display choice
+        <div className='outer'>
+          <input className='fade' type="button" onClick={this.onButtonClick} value="Pick your puck"/>
```

Comment:
Out of curiosity, why does `this.onButtonClick` not require a bind? It uses `this.setState` which makes me think it would, though I do get how it's working nonetheless